examples/quickstart/CMakeLists.txt

```diff
@@ -9,19 +9,29 @@
 # handle such niceties. On a case-by-case basis it is fixable but it requires work that CMake knows how
 # to do but that is not trivial.
 #
-IF(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
+
+#
+# On a system like macOS, ${CMAKE_CXX_COMPILER} evaluates to something that is not c++,
+# but rather another program that requires -isysroot=... to be functional. To achieve
+# compilation, you have to build the -isysroot=... command, when it is needed. This is
+# normally handled automatically. It is probably not unique to macOS. Other systems like
+# Android might require similar hand-holding.
+#
+find_program(CPPCOMMAND c++ PATHS /usr/bin /usr/local/bin)
```

Comment:
That seems insane! Man, systems that force people to learn customization options and hidden arguments before they can even **try** to work with them really boil my blood.

Reply:
Well, on a mac, you can just do "c++ myfile.cpp". This calls, somewhat, the compiler with the proper library. Apple puts the library (include and so forth) *off* of root. So you can actually switch or have various versions at the same time. It is neat. And all transparent.

Yet, for some reason, CMake goes directly, underneath, so that it has to specify the sysroot and everything. I am sure that there must be a good reason for it.